docs/topics/forms/index.txt

```diff
@@ -733,12 +733,17 @@ Reusable form templates
 
 If your site uses the same rendering logic for forms in multiple places, you
 can reduce duplication by saving the form's loop in a standalone template and
-using the :ttag:`include` tag to reuse it in other templates:
+using over ride your forms ``__str__`` method to reuse it in other templates:
 
 .. code-block:: html+django
 
-    # In your form template:
-    {% include "form_snippet.html" %}
+    # In your template:
+    {{ form }}
+
+    # In your form:
+    class MyForm(forms.Form):
+        def __str__(self):
+            return self.render(template_name='form_snippet.html')
```

Comment:
...following on from that, we shouldn't encourage overriding `.__str__()` as that already calls `.render()` which will use `.template_name`. So this should simply be:
```suggestion
    # In your template:
    {{ form }}

    # In your form:
    class MyForm(forms.Form):
        template_name = 'form_snippet.html'
```

Unless I'm mistaken?

(Although this example doesn't connect the fact that `MyForm` will be instantiated and passed in the context... Also the highlighting will be off due to mixing `html+django` with `python` content.)

Reply:
I'll have a think again about this section.

You are quite right generally, we should be encouraging overriding `template_name`. I thought I had updated for this but clearly not!